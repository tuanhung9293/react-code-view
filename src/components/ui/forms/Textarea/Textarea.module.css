@value colors: 'styles/colors.css';
@value field, separator, error, secondary from colors;

.textarea{
  appearance: none;
  border: 2px solid transparent;
  background-color: field;
  border-radius: 4px;
  box-shadow: 0 2px rgba(0,0,0,0.1);
  margin: 10px 0;
  padding: 10px 15px;
  font-family: 'campton', Helvetica, sans-serif;
  line-height: 1.5rem;
  width: 100%;
  box-sizing: border-box;
  height: 200px;
  resize: none;
}

.error{
  border-color: error;
}

.textarea:focus{
  outline: 0;
  border-color: secondary;
}
