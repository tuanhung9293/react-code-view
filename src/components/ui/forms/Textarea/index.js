import React from 'react'
import classNames from 'classnames'
import styles from './Textarea.module.css'

function Textarea({hasError, value, ...rest}) {
  const className = classNames({
    [styles.textarea]: true,
    [styles.error]: hasError
  })
  return (
    <textarea {...rest} value={value || ''} className={className} />
  )
}

export default Textarea