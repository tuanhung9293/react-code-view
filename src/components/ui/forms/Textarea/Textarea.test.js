import React from 'react'
import {shallow} from 'enzyme'
import Textarea from './'


describe('renders', () => {
  it('renders Textarea', () => {
    expect(shallow(
        <Textarea />
    )).toMatchSnapshot()
  })

  it('renders Textarea with props hasError', () => {
    expect(shallow(
        <Textarea hasError />
    )).toMatchSnapshot()
  })
})
