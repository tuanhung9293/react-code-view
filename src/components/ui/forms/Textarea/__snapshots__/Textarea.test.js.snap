// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders Textarea 1`] = `
<textarea
  className="textarea"
  value=""
/>
`;

exports[`renders renders Textarea with props hasError 1`] = `
<textarea
  className="textarea error"
  value=""
/>
`;
